File: Hot4/Hot4.Server/Brands/NetworkBase.cs
//using Hot4.Core.Helper;
//using Hot4.DataModel.Models;
//using Hot4.Service.Abstract;
//using Hot4.ViewModel;
//using Microsoft.Data.SqlClient;

//namespace Hot4.Server.Brands
//{
//    public abstract class NetworkBase
//    {
//        protected string ApplicationName { get; set; }
//        protected SqlConnection SqlConn { get; set; }
//        protected string ServiceEndpoint { get; set; }
//        public bool IsTestMode { get; set; }
//        public bool WebServiceOrCorporate { get; set; }
//        public string ReferencePrefix { get; set; }

//        protected NetworkBase(SqlConnection sqlConn, string applicationName, string serviceEndpoint, bool isTestMode, bool webServiceOrCorporate, string referencePrefix = "999")
//        {
//            ApplicationName = applicationName;
//            SqlConn = sqlConn;
//            ServiceEndpoint = serviceEndpoint;
//            IsTestMode = isTestMode;
//            WebServiceOrCorporate = webServiceOrCorporate;
//            ReferencePrefix = referencePrefix;
//        }

//        public virtual string GetReference(string mobile)
//        {
//            return ReferencePrefix + "+" + mobile + "+" + DateTime.Now.ToString("dd-MM-yyyy HH:mm:ss.fff");
//        }

//        protected void ApplyDiscountRules(RechargeModel rechargeModel)
//        {
//            if (!WebServiceOrCorporate)
//            {
//                var limit = new LimitDiscountTo5Percent();
//                limit.Apply(rechargeModel);
//            }
//        }
//        public RechargePrepaid CreateRechargeObject(RechargeModel  rechargeModel)
//        {
//            var iRechargePrepaid = new RechargePrepaid
//            {
//                RechargeId = rechargeModel.RechargeId,
//                DebitCredit = rechargeModel.Amount >= 0,
//                Reference = GetReference(rechargeModel.Mobile),
//                FinalBalance = -1,
//                InitialBalance = -1,
//                Narrative = "", // Explicit ,ade empty
//                ReturnCode = "-1" // Explicit change into string 
//            };
//            iRechargePrepaid.Narrative = iRechargePrepaid.DebitCredit ? "Pending" : "Debit Pending";
//            return iRechargePrepaid;
//        }

//        public abstract ServiceRechargeResponse RechargePrepaid(Recharge recharge, int numberOfDays = 90);
//        public abstract INetworkClient GetNetworkClient(int timeout = 30000);
//    }
//}

File: Customer APIs/hot.backend.all/hot.core/hot.core/Brands/NetworkBase.vb
Imports System.Data.SqlClient
Imports Hot.Core.ServiceWrappers
Imports Hot.Data

Namespace Brands
    Public MustInherit Class NetworkBase(Of T)

        Protected ApplicationName As String
        Protected SqlConn As SqlConnection
        Protected ServiceEndpoint As String
        Protected MustOverride Function Name As String
        Public Property IsTestMode As Boolean
        Public Property WebServiceOrCorporate As Boolean
        Public Property ReferencePrefix As String
        
        Public Sub New(sqlConn As SqlConnection, applicationName As String, serviceEndpoint As String, isTestMode As Boolean, webServiceOrCorporate As Boolean, Optional referencePrefix As String = "999")
            Me.ApplicationName = applicationName
            Me.SqlConn = sqlConn
            Me.ServiceEndpoint = serviceEndpoint
            Me.IsTestMode = isTestMode
            Me.WebServiceOrCorporate = webServiceOrCorporate
            Me.ReferencePrefix = referencePrefix
        End Sub

        Public Overridable Function GetReference(mobile As String) As String
            Return ReferencePrefix & "+" & mobile & "+" & Format(Now, "dd-MM-yyyy HH:mm:ss.fff")
        End Function

        Protected Sub ApplyDiscountRules(recharge As xRecharge)
            If Not WebServiceOrCorporate then
                Dim limit = new LimitDiscountTo5Percent()
                limit.Apply(recharge)
            End If
        End Sub

        Public Function CreateRechargeObject(recharge As xRecharge) As xRechargePrepaid
            Dim iRechargePrepaid As New xRechargePrepaid
            iRechargePrepaid.RechargeID = recharge.RechargeID
            iRechargePrepaid.DebitCredit = recharge.Amount >= 0
            iRechargePrepaid.Reference = GetReference(recharge.Mobile)
            iRechargePrepaid.FinalBalance = - 1
            iRechargePrepaid.InitialBalance = - 1
            iRechargePrepaid.Narrative = IIf(iRechargePrepaid.DebitCredit, "Pending", "Debit Pending")
            iRechargePrepaid.ReturnCode = - 1
            Return iRechargePrepaid
        End Function

        Public MustOverride Function RechargePrepaid(recharge As xRecharge, Optional numberOfDays As Integer = 90) As ServiceRechargeResponse

        Public MustOverride Function GetNetworkClient(Optional timeout As Integer = 30000) As INetworkClient(Of T)
    End Class
End Namespace